mne/label.py

```diff
@@ -897,6 +912,110 @@ def write_label(filename, label, verbose=None):
+    """Helper to get label and subject information prior to label spliting"""
+
+    # If necessary, find the label
+    if isinstance(label, BiHemiLabel):
+        raise TypeError("Can only split labels restricted to one hemisphere.")
+    elif isinstance(label, string_types):
+        label = read_label(label)
+
+    # Find the subject
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    if label.subject is None and subject is None:
+        raise ValueError("The subject needs to be specified.")
+    elif subject is None:
+        subject = label.subject
+    elif label.subject is None:
+        pass
+    elif subject != label.subject:
+        raise ValueError("The label specifies a different subject (%r) from "
+                         "the subject parameter (%r)."
+                         % label.subject, subject)
+
+    return label, subject, subjects_dir
+
+
+def _split_label_contig(label_to_split, subject=None, subjects_dir=None):
+    """Split label into contiguous regions (i.e., connected components)
+
+    Parameters
+    ----------
+    label_to_split : Label | str
+        Label which is to be split (Label object or path to a label file).
+    subject : None | str
+        Subject which this label belongs to (needed to locate surface file;
+        should only be specified if it is not specified in the label).
+    subjects_dir : None | str
+        Path to SUBJECTS_DIR if it is not set in the environment.
+
+    Returns
+    -------
+    labels : list of Label
+        The contiguous labels, in order of decending size.
+    """
+    # Convert to correct input if necessary
+    label_to_split, subject, subjects_dir = _prep_label_split(label_to_split,
+                                                              subject,
+                                                              subjects_dir)
+
+    # Find the spherical surface to get vertices and tris
+    surf_fname = '.'.join((label_to_split.hemi, 'sphere'))
+    surf_path = op.join(subjects_dir, subject, 'surf', surf_fname)
+    surface_points, surface_tris = read_surface(surf_path)
+
+    # Get vertices we want to keep
+    verts_arr = label_to_split.vertices
+    # Compute edges on only upper triangle (since edges are binary)
+    edges_all = sparse.triu(mesh_edges(surface_tris), format='csr')
+
+    # Subselect rows and cols of vertices that belong to the label
+    select_edges = edges_all[verts_arr].tocsc()[:, verts_arr].tocsr()
+
+    # Compute connected components and store as lists of vertex numbers
+    n_comp, comp_labels = connected_components(select_edges, directed=False)
+    label_divs = []
+    for label_id in range(n_comp):
+        label_divs.append(verts_arr[comp_labels == label_id])
+
+    # Construct label division names
+    n_parts = len(label_divs)
+    if label_to_split.name.endswith(('lh', 'rh')):
+        basename = label_to_split.name[:-3]
+        name_ext = label_to_split.name[-3:]
```

Comment:
The ends have `-` characters instead of dots, so I can't quite do it with `splitext`. Is there another way that's more pythonic?

```
>>>op.splitext(label_DMN.name)
('lh', '.7Networks_7-lh')

>>>label_DMN.name[:-3]
'lh.7Networks_7'
>>>label_to_split.name[-3:]
'-lh'
```